### PR TITLE
fix(jdbc): support fractional seconds in getTime

### LINKED ISSUE #10682: JDBC ResultSet.getTime() returns null if time value has fractional seconds
### What happens?

I cannot fetch `ResultSet.getTime()` from the JDBC driver.

### To Reproduce

Try this Java code:

```java
import org.duckdb.DuckDBConnection;
import java.sql.DriverManager;

DuckDBConnection connection = (DuckDBConnection) DriverManager.getConnection("jdbc:duckdb:");
try (
    Statement s = connection.createStatement();
    ResultSet rs = s.executeQuery("select current_time")
) {
    while (rs.next()) {
        System.out.println(rs.getTime(1));
        System.out.println(rs.getString(1));
    }
}
```

It prints something along the lines of:

```
null
09:32:52.936
```

The problem is that `java.sql.Time.valueOf(String)` doesn't accept fractional seconds. After the exception is thrown by JDBC, `null` is simply returned:

https://github.com/duckdb/duckdb/blob/21389fa53053fb36c4d37d922e34041b46969576/tools/jdbc/src/main/java/org/duckdb/DuckDBVector.java#L307

### OS:

Microsoft Windows [Version 10.0.22631.3007]

### DuckDB Version:

0.10.0

### DuckDB Client:

JDBC

### Full Name:

Lukas Eder

### Affiliation:

Data Geekery

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Mause**
Also cleans up some TODO's I left a while back

Fixes #10682

### CODE REVIEW

**Review by hawkfish — APPROVED** [review]
TIME_TZ is not horribly important but we should make sure it works before 1.0.

File: tools/jdbc/src/main/java/org/duckdb/DuckDBVector.java [hunk 1]
```diff
@@ -124,13 +125,31 @@ Object getObject(int idx) throws SQLException {
         }
     }
 
-    LocalTime getLocalTime(int idx) throws SQLException {
+    LocalTime getLocalTime(int idx) {
+        if (check_and_null(idx)) {
+            return null;
+        }
+
+        if (isType(DuckDBColumnType.TIME)) {
```
[the remark this answers is not in the document]
Comment (Mause) [reply tools/jdbc/src/main/java/org/duckdb/DuckDBVector.java:133]: I don't think it really makes sense to handle TIME_TZ in the getLocalTime method?

File: tools/jdbc/src/jni/duckdb_java.cpp [hunk 2]
```diff
@@ -844,6 +844,12 @@ jobject ProcessVector(JNIEnv *env, Connection *conn_ref, Vector &vec, idx_t row_
 	case LogicalTypeId::DOUBLE:
 		constlen_data = env->NewDirectByteBuffer(FlatVector::GetData(vec), row_count * sizeof(double));
 		break;
+	case LogicalTypeId::DATE:
+		constlen_data = env->NewDirectByteBuffer(FlatVector::GetData(vec), row_count * sizeof(date_t));
+		break;
+	case LogicalTypeId::TIME:
+		constlen_data = env->NewDirectByteBuffer(FlatVector::GetData(vec), row_count * sizeof(dtime_t));
+		break;
 	case LogicalTypeId::TIME_TZ:
```
[the remark this answers is not in the document]
Comment (Mause) [reply tools/jdbc/src/jni/duckdb_java.cpp:853]: Fixed

**Review by hawkfish — APPROVED** [review]

**Comment by Mytherin** [comment]
Thanks

**Pull request merged**